src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -174,12 +174,12 @@ private void SetGlobalErrorIfTrue(bool arg)
             if (filterOpt == null)
             {
                 WarnUnusedFields(compilation, diagnostics, cancellationToken);
-            }
 
-            MethodSymbol entryPoint = GetEntryPoint(compilation, moduleBeingBuiltOpt, hasDeclarationErrors, diagnostics, cancellationToken);
-            if (moduleBeingBuiltOpt != null && entryPoint != null && compilation.Options.OutputKind.IsApplication())
-            {
-                moduleBeingBuiltOpt.SetPEEntryPoint(entryPoint, diagnostics);
+                MethodSymbol entryPoint = GetEntryPoint(compilation, moduleBeingBuiltOpt, hasDeclarationErrors, diagnostics, cancellationToken);
```

Comment:
```none
'{0}': an entry point cannot be marked with the 'async' modifier
Program does not contain a static 'Main' method suitable for an entry point
Program has more than one entry point defined. Compile with /main to specify the type that contains the entry point.
'{0}' has the wrong signature to be an entry point
'{0}': an entry point cannot be generic or in a generic type
Could not find '{0}' specified for Main method
'{0}' specified for Main method must be a valid non-generic class or struct
'{0}' does not have a suitable static Main method
Cannot use '{0}' for Main method because it is imported
```

plus a couple of errors related to main appearing in a script.